RecoBTag/FeatureTools/plugins/DeepDoubleXTagInfoProducer.cc

```diff
@@ -137,11 +140,15 @@ void DeepDoubleBTagInfoProducer::produce(edm::Event& iEvent,
     {
 
         // create data containing structure
-        btagbtvdeep::DeepDoubleBFeatures features;
+        btagbtvdeep::DeepDoubleXFeatures features;
 
         // reco jet reference (use as much as possible)
         const auto& jet = jets->at(jet_n);
+
         edm::RefToBase<reco::Jet> jet_ref(jets, jet_n);
+	if (jet.pt() > min_jet_pt_)
```

Comment:
which pt are we using at this point?
I printed out a few cases before the last changes and I see the discriminator value going to 0 for jets with pt above 150 GeV (the values below are slimmedJetsAK8 jet pt() [corrected value]).
My guess is that the pt at this point is not corrected. Is this choice OK for your physics needs?

e.g. in wf 136.8311 
```
 loaded event 54880615
jet 0 170.957 discr 10 pfMassIndependentDeepDoubleCvBJetTags:probHbb 0                 <== was 0.324633

 loaded event 54936747
jet 2 150.022 discr 13 pfMassIndependentDeepDoubleCvLJetTags:probHcc 0                 <== was 1

```

Reply:
Correct, IIUC this is the uncorrected pt. The 150 cut is OK, because I think we're only saving jets with pT > 170 and for our analysis we need higher pT jets. The concern is for other analyses but I think at the 150 anything 200 is safe.

Reply:
The cut is not at 170 but 100 GeV. But because of the loose cut, the jet pT can also go below 100 GeV (https://github.com/cms-sw/cmssw/blob/CMSSW_9_4_10/PhysicsTools/PatAlgos/python/slimming/applySubstructure_cff.py#L186-L188).

Reply:
I think 150 GeV on uncorrected pT should be fine because, as you said, meaninful analysis level cuts will be quite higher than that. I was just pointing out that the cut on ak8 puppi jets stored in MiniAOD is not a flat 170 GeV cut but is actually lower and a bit more complex because of the extra `cutLoose` and `nLoose` requirements. What will be done with jets that do not pass the cut? Some dummy discriminator value (preferably negative) will be stored?

Reply:
Only limited information is stored for lower-pT AK8 jets in miniAOD. @rappoccio can clarify further

Reply:
The default value when the output is initialized is 0, we could make it -1 I guess?

Reply:
@kpedro88 I guess you are referring to the 170 GeV on CHS fat jets. Yes, only some of the Puppi fat jets (generally those with high pT) will have a matched CHS fat jets from which kinematic and other variables are picked up and added to the Puppi jet as userFloat.

However, here I am discussing something different since btag discriminators are added in a completely different way from userFloats so the above-mentioned cut on CHS jets plays no role. Here we are not so much after reducing the event size (although, presumably a lot of identical dummy discriminator values stored in the event can lead to better event compression) but reducing the processing time.

Reply:
@andrzejnovak I think a negative value would be better than some other normally allowed value. That way it will be easier to discriminate skipped jets based on their discriminator value.